zerver/webhooks/facebook/view.py

```diff
@@ -0,0 +1,114 @@
+    message = '**{field} permission** changed'.format(field=field)
+    if field == 'ads_management':
+        message = message + '\n* {verb}'.format(
+            verb=payload['entry'][0]['changes'][0]['value']['verb']
+        )
+        for id in payload['entry'][0]['changes'][0]['value']['target_ids']:
+            message = message + '\n  * {id}'.format(id=id)
+    elif field == 'manage_pages':
+        message = message + '\n* {verb}'.format(
+            verb=payload['entry'][0]['changes'][0]['value']['verb']
+        )
+        for id in payload['entry'][0]['changes'][0]['value']['target_ids']:
+            message = message + '\n  * {id}'.format(id=id)
+    return message
+
+def application_object(payload: Dict[Text, Any]) -> Text:
+    field = payload['entry'][0]['changes'][0]['field']
+    message = '**{field}** received'.format(field=field)
+    if field == 'plugin_comment':
+        message = message + '\n**{msg_user}:**\n```quote\n{message}\n```'.format(
+            msg_user=payload['entry'][0]['changes'][0]['value']['from']['name'],
+            message=payload['entry'][0]['changes'][0]['value']['message']
+        )
+    if field == 'plugin_comment_reply':
+        message = message + '\n**{prt_msg_user}:** (Parent)\n'\
+            '```quote\n{prt_message}\n```'.format(
+                prt_msg_user=payload['entry'][0]['changes'][0]['value']['parent']['from']['name'],
+                prt_message=payload['entry'][0]['changes'][0]['value']['parent']['message']
+            )
+        message = message + '\n**{cld_msg_user}:**\n```quote\n'\
+            '```quote\n{cld_message}\n```\n```'.format(
+                cld_msg_user=payload['entry'][0]['changes'][0]['value']['from']['name'],
+                cld_message=payload['entry'][0]['changes'][0]['value']['message']
+            )
+    return message
+
+@api_key_only_webhook_view("Facebook")
+@has_request_variables
+def api_facebook_webhook(request: HttpRequest, user_profile: UserProfile,
+                         stream: Text=REQ(default='Facebook'), token: Text=REQ()) -> HttpResponse:
+
+    if request.method == 'GET':  # facebook webhook verify
+        if request.GET.get("hub.mode") == 'subscribe':
+            if request.GET.get('hub.verify_token') == token:
+                return HttpResponse(request.GET.get('hub.challenge'))
+            else:
+                return json_error(('Error: Token is wrong'))  # nocoverage
+        else:
+            return json_error(('Error: Unsupported method'))  # nocoverage
+    else:  # facebook webhook receiver
+        payload = json.loads(request.body.decode("UTF-8"))
+        object = get_object(payload)
+        if object is not None:
+            body = get_body_based_on_object(object)(payload)
+            subject = object + " notification"
+            check_send_stream_message(user_profile, request.client,
+                                      stream, subject, body)
+        return json_success()
```

Comment:
I would use `event` instead of `object` and `get_event` instead of `get_object` for two reasons:
* `object` is a Python keyword, so we should probably never use it.
* `event` and `get_event` are more consistent with how similar functions are named in our other webhook implementations.